read-fonts/src/tables/variations.rs

```diff
@@ -850,11 +884,180 @@ where
     ///
     /// This does not account for scaling. Returns only explicitly encoded
     /// deltas, e.g. an omission by IUP will not be present.
-    pub fn deltas(&'a self) -> TupleDeltaIter<'a, T> {
-        TupleDeltaIter::new(&self.point_numbers, &self.packed_deltas)
+    pub fn deltas(&self) -> TupleDeltaIter<'a, T> {
+        let (point_numbers, packed_deltas) = self.point_numbers_and_packed_deltas();
+        let count = point_numbers.count() as usize;
+        let packed_deltas = if count == 0 {
+            PackedDeltas::consume_all(packed_deltas)
+        } else {
+            PackedDeltas::new(packed_deltas, if T::is_point() { count * 2 } else { count })
+        };
+        TupleDeltaIter::new(&point_numbers, packed_deltas)
+    }
+
+    fn point_numbers_and_packed_deltas(&self) -> (PackedPointNumbers<'a>, FontData<'a>) {
+        if self.header.tuple_index().private_point_numbers() {
+            PackedPointNumbers::split_off_front(self.serialized_data)
+        } else {
+            (
+                self.shared_point_numbers.clone().unwrap_or_default(),
+                self.serialized_data,
+            )
+        }
     }
 }
 
+impl TupleVariation<'_, GlyphDelta> {
+    pub fn accumulate_dense_deltas<D: PointCoord>(
```

Comment:
If it's public API I'd love some docs 👼

Reply:
oops, I totally meant to write docs for these before requesting review. done now :)